rainier-cats/src/main/scala/com/stripe/rainier/cats/package.scala

```diff
@@ -4,16 +4,34 @@ import com.stripe.rainier.compute.Real
 import com.stripe.rainier.core.Generator
 import com.stripe.rainier.core.RandomVariable
 import com.stripe.rainier.sampler.RNG
-import _root_.cats.Monad
+import _root_.cats.{Comonad, Monad}
 
 import scala.annotation.tailrec
 
-object `package` {
+object `package` extends LowPriorityInstances {
   implicit val rainierMonadGenerator: Monad[Generator] = new MonadGenerator
   implicit val rainierMonadRandomVariable: Monad[RandomVariable] =
     new MonadRandomVariable
 }
 
+private[cats] sealed abstract class LowPriorityInstances {
+  implicit def generatorComonad(implicit r: RNG,
+                                n: Numeric[Real]): Comonad[Generator] =
+    new MonadGenerator with Comonad[Generator] {
```

Comment:
I could ALSO have just changed the monad instance to extend `BiMonad` and added `coflatMap` and `extract` directly to the old class... but I didn't want to break binary compatibility.